### PR TITLE
Added Central Logger

### PR DESCRIPTION
**Pull request by sksinha2410**
Added Central Logger to streamline application logging. Implemented logging with Firebase Firestore for centralized log management.

### CODE REVIEW

**Comment by sksinha2410** [comment]
If possible then delete my all branches after merging

**Review by saurabhalp — APPROVED** [review]

File: app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.starstruckstech.pgpodandroid
+
+import com.google.firebase.firestore.FirebaseFirestore
+
+object CentralLogger {
+    private val db = FirebaseFirestore.getInstance()
+
+    fun log(level: LogLevel, tag: String, message: String) {
+        val logEntry = LogEntry(level, tag, message)
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt:9]: This should be implemented using Timber. You need to have 3 functions for now :
logWarning
logError
logStatement

File: app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.starstruckstech.pgpodandroid
+
+import com.google.firebase.firestore.FirebaseFirestore
+
+object CentralLogger {
+    private val db = FirebaseFirestore.getInstance()
+
+    fun log(level: LogLevel, tag: String, message: String) {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt:8]: add java docs

File: app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.starstruckstech.pgpodandroid
+
+import com.google.firebase.firestore.FirebaseFirestore
+
+object CentralLogger {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/CentralLogger.kt:5]: Java Docs

File: app/src/main/java/com/starstruckstech/pgpodandroid/LogAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.starstruckstech.pgpodandroid
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class LogAdapter(private val logs: List<LogEntry>) : RecyclerView.Adapter<LogAdapter.LogViewHolder>() {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/LogAdapter.kt:9]: If this is for testing purpose move it to a new directory named `TestingUtility` or remove this

File: app/src/main/java/com/starstruckstech/pgpodandroid/LogLevel.kt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package com.starstruckstech.pgpodandroid
+
+enum class LogLevel {
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/LogLevel.kt:3]: java docs

File: app/src/main/java/com/starstruckstech/pgpodandroid/LogEntry.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.starstruckstech.pgpodandroid
+
+data class LogEntry(
```
Comment (Jha2508) [inline app/src/main/java/com/starstruckstech/pgpodandroid/LogEntry.kt:3]: same as above

**Pull request merged**